PubSub/src/Subscription.php

```diff
@@ -722,10 +729,19 @@ public function acknowledgeBatch(array $messages, array $options = [])
     {
         $this->validateBatch($messages, Message::class);
 
-        $this->connection->acknowledge($options + [
-            'subscription' => $this->name,
-            'ackIds' => $this->getMessageAckIds($messages)
-        ]);
+        // the rpc may throw errors for a sub with EOD enabled
+        // but we don't act on the exception to maintain compatibility
+        try {
```

Comment:
I see we have system tests for this, but would love to see unit as well if possible.